src/test/kotlin/org/jetbrains/research/tasktracker/tracking/TaskDocumentListenerTest.kt

```diff
@@ -0,0 +1,74 @@
+package org.jetbrains.research.tasktracker.tracking
+
+import com.intellij.openapi.application.ReadAction
+import com.intellij.openapi.command.WriteCommandAction
+import com.intellij.openapi.editor.Document
+import com.intellij.openapi.fileEditor.FileDocumentManager
+import com.intellij.testFramework.fixtures.BasePlatformTestCase
+import org.jetbrains.research.tasktracker.config.MainTaskTrackerConfig
+import org.jetbrains.research.tasktracker.tracking.logger.DocumentLogger
+import java.io.File
+
+class TaskDocumentListenerTest : BasePlatformTestCase() {
+
+    fun testTaskDocumentListener() {
+        val (logFile, document) = prepareDocument()
+        linesToWrite.forEach {
+            document.writeText(it)
+        }
+        val text = linesToWrite.joinToString(separator = "")
+        ReadAction.run<Exception> {
+            assert(document.text == text) { "Expected '$text' to be written, but found '${document.text}'" }
+        }
+        document.reWriteText("test")
+        DocumentLogger.getDocumentLogPrinter(document)?.getActiveLogPrinter(document)?.csvPrinter?.flush()
+        assert(logFile.exists()) { "log file with path '${logFile.path}' should have been created" }
+        val changes = logFile.readLines().map {
+            it.split(",")[5]
+                .replace("\"", "")
+        }.takeLast(3) // skip
```

Comment:
It looks like magic. Please, add some comments - why do you use 5, tale last 3 etc